### PR TITLE
add migration guide for profile

### PR DESCRIPTION
**Pull request by tadelesh**
Resolve: #20462

### CODE REVIEW

File: documentation/profile-migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Guide for migrating from `profiles/**`
+
+This document is intended for users that are familiar with the previous version of the Azure SDK For Go profiles package (`github.com/Azure/azure-sdk-for-go/profiles/**`) and wish to migrate their application to the next version of Azure SDK for Go.
```
Comment (josefree) [inline documentation/profile-migration-guide.md:3]: ... the older versions of Azure Profile SDK module for Go, and target to migrate applications to the latest version of the same module.

File: documentation/profile-migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- [Package mapping](#package-mapping)
+- [API changes](#API-changes)
+
+## Prerequisites
+
+- Go 1.18
+
+## Package mapping
+
+There are several sub-packages under pervious version of the Azure SDK For Go profiles package (`profiles/**`): `2017-03-09`, `2018-03-01`, `2019-03-01`, `2020-09-01`, `latest`, `preview`.
```
Comment (josefree) [inline documentation/profile-migration-guide.md:17]: There are several released packages of Azure Profile SDK for Go: ...

File: documentation/profile-migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+For those who are using `latest` or `preview`, you need to migrate to our latest modules under `github.com/Azure/azure-sdk-for-go/sdk`. The latest SDK only provides one API version in one module for each RP. If you are using several RP's sub-packages, you need to migrate to several modules. You can search in [pkg.go.dev](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go) to find all the stable and preview version of all RPs.
```
Comment (josefree) [inline documentation/profile-migration-guide.md:19]: ”The latest SDK only provides one API version in one module for each RP. If you are using several RP's sub-packages, you need to migrate to several modules. “ a little difficult to understand this.  The proposal according to my understanding:

... However, the modules of different resource providers are not packaged all in one.  Only the module(s) of the same resource provider is(are) packaged together. To migrate to the latest modules, you need to find all packages that contain required modules.  You can search in pkg.go.dev to find those resource providers and new releases.

File: documentation/profile-migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+For example, if you are using `github.com/Azure/azure-sdk-for-go/profiles/latest/containerservice/mgmt/containerservice`, then you need to migrate to `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice@v2.4.0`; if you are using `github.com/Azure/azure-sdk-for-go/profiles/preview/containerservice/mgmt/containerservice`, then you need to migrate to `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice@2.5.0-beta.1`.
+
+For those who are using `2017-03-09`, `2018-03-01` or `2019-03-01`, unfortunately, we do not provide these old profile version in our latest Azure SDK for Go. Please upgrade to the latest profile `2020-09-01` first.
```
Comment (josefree) [inline documentation/profile-migration-guide.md:23]: For those who are using `2017-03-09`, `2018-03-01` or `2019-03-01`,  there is no direct mapping in the new releases.  You have to upgrade to the Azure Profile module of `2020-09-01` first.

File: documentation/profile-migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+For those who are using `2020-09-01`, you need to migrate to our latest profile module `github.com/Azure/azure-sdk-for-go/profile/p20200901`.
+
+## API changes
+
+For `management` modules, you could refer [this](https://aka.ms/azsdk/go/mgmt/migration) for migrations. For `client` modules, you could refer README.md file of cooresponding module.
```
Comment (josefree) [inline documentation/profile-migration-guide.md:29]: For `management` modules, you could refer to this document for migration. For `client` modules, you could refer to "Client modules" section in the [README.md](https://github.com/Azure/azure-sdk-for-go/blob/main/README.md#client-modules) file.

File: documentation/profile-migration-guide.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+## Need help?
+
+If you have encountered an issue during migration, please file an issue via [Github Issues](https://github.com/Azure/azure-sdk-for-go/issues) and make sure you add the "Preview" label to the issue
```
Comment (josefree) [inline documentation/profile-migration-guide.md:33]: If you encounter any issues in migration, please file an issue [here](https://github.com/Azure/azure-sdk-for-go/issues), and make sure to add "Preview" label with the issue.

Comment (josefree) [reply documentation/profile-migration-guide.md:33]: why "preview" label? why not "migration" label?

**Review by josefree — CHANGES_REQUESTED** [review]
see in previous comments

File: documentation/profile-migration-guide.md (hunk 1 above)
Comment (tadelesh) [reply documentation/profile-migration-guide.md:3]: The previous version does not use module. I changed some words accordingly.

File: documentation/profile-migration-guide.md (hunk 3 above)
Comment (tadelesh) [reply documentation/profile-migration-guide.md:19]: Changed some expressions.

File: documentation/profile-migration-guide.md (hunk 6 above)
Comment (tadelesh) [reply documentation/profile-migration-guide.md:33]: From legacy doc. Changed both.

File: documentation/profile-migration-guide.md (hunk 4 above)
Comment (tadelesh) [reply documentation/profile-migration-guide.md:23]: Changed some expressions.

**Review by josefree — APPROVED** [review]

**Pull request merged**